src/game_state/player.rs

```diff
@@ -72,186 +49,172 @@ pub struct ReportMessage {
-                        yaw: float_to_angle(msg.new_angle.yaw),
-                        pitch: float_to_angle(msg.new_angle.pitch),
-                        on_ground: false,
-                    }),
-                    Some(player.conn_id),
-                    true
+                    msg.conn_id,
+                    Packet::SpawnPlayer(player.spawn_player_packet())
                 )
                 .unwrap();
-                player_clone.position = msg.new_position;
-                player_clone.angle = msg.new_angle;
-                players.insert(msg.conn_id, player_clone);
-            }
-            PlayerStateOperations::Report(msg) => {
-                players.iter().for_each(|(conn_id, player)| {
-                    if *conn_id != msg.conn_id {
-                        let player_clone = player.clone();
-                        send_packet!(
-                            messenger,
-                            msg.conn_id,
-                            Packet::PlayerInfo(PlayerInfo {
-                                action: 0,
-                                number_of_players: 1, //send each player in an individual packet for now
-                                uuid: player_clone.uuid.as_u128(),
-                                name: player_clone.name.clone(),
-                                number_of_properties: 0,
-                                gamemode: 1,
-                                ping: 100,
-                                has_display_name: false,
-                            })
-                        )
-                        .unwrap();
-                        send_packet!(
-                            messenger,
-                            msg.conn_id,
-                            Packet::SpawnPlayer(SpawnPlayer {
-                                entity_id: player.entity_id,
-                                uuid: player_clone.uuid.as_u128(),
-                                x: player_clone.position.x,
-                                y: player_clone.position.y,
-                                z: player_clone.position.z,
-                                yaw: 0,
-                                pitch: 0,
-                                entity_metadata_terminator: 0xff,
-                            })
-                        )
-                        .unwrap();
-                    }
-                })
             }
+        }),
+    }
+}
+
+impl Player {
+    pub fn move_and_look(
+        &mut self,
+        new_position: Option<Position>,
+        new_angle: Option<Angle>,
+    ) -> EntityLookAndMove {
+        if let Some(new_angle) = new_angle {
+            self.angle = new_angle;
+        }
+        let update_packet = self.entity_look_and_move_packet(new_position);
+        if let Some(new_position) = new_position {
+            self.position = new_position;
+        }
+        update_packet
+    }
+
+    pub fn join_game_packet(&self) -> JoinGame {
```

Comment:
This is a new pattern I'm trying out- all methods that generate packets centered around a given entity should be generated by that entity. In this case, I think it definitely looks more organized and separates concerns.